lib/ask/runtime/broker.ex

```diff
@@ -115,6 +115,11 @@ defmodule Ask.Runtime.Broker do
       reached_quotas = reached_quotas?(survey)
       survey_completed = survey.cutoff <= completed || reached_quotas
       batch_size = batch_size(survey, by_state)
+      survey = if survey.first_window_started_at do
+        survey
+      else
+        Survey.changeset(survey, %{first_window_started_at: SystemTime.time.now}) |> Repo.update!
+      end
```

Comment:
I would extract this to a separate function for better understanding. I'm thingking of something like
```elixir
def update_survey_on_poll(survey)
```
what do you think?